src/smvlang/smv_range.h

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************\
+
+Module: SMV Typechecking
+
+Author: Daniel Kroening, dkr@amazon.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_SMV_RANGE_H
+#define CPROVER_SMV_RANGE_H
+
+#include <util/arith_tools.h>
+
+class smv_ranget
+{
+public:
+  smv_ranget() : from(0), to(0)
+  {
+  }
+
+  smv_ranget(mp_integer _from, mp_integer _to)
+    : from(std::move(_from)), to(std::move(_to))
+  {
+  }
+
+  mp_integer from, to;
+
+  bool is_contained_in(const smv_ranget &other) const
+  {
+    if(other.from > from)
+      return false;
+    if(other.to < to)
+      return false;
+    return true;
+  }
+
+  void make_union(const smv_ranget &other)
+  {
+    mp_min(from, other.from);
+    mp_max(to, other.to);
+  }
+
+  void to_type(typet &dest) const
+  {
+    dest = typet(ID_range);
+    dest.set(ID_from, integer2string(from));
+    dest.set(ID_to, integer2string(to));
+  }
+
+  bool is_bool() const
+  {
+    return from == 0 && to == 1;
```

Comment:
While I understand that this is the same behaviour as before: why would from == 1 or to == 0 not be permitted?

Reply:
See #903, and will add a precondition to the class.